### ISSUE TITLE
Inside the `where` object, the `NOT` field must be last or you can get unexpected results

### ISSUE THREAD

**Issue by ehicks05** [prisma/prisma#23897]
### Bug description

Let's say you have this schema:
```prisma
model BugReproModel {
  id String @id @default(auto()) @map("_id") @db.ObjectId

  topic String
  tag   String
}
```

and this data:
```json
[
  { "id": "65e2051871e92cf2cffdf229", "topic": "a", "tag": "d" },
  { "id": "65e2051871e92cf2cffdf22a", "topic": "b", "tag": "e" },
  { "id": "65e2051971e92cf2cffdf22b", "topic": "c", "tag": "f" }
]
```

My team was recently confused when two queries that only differ in field ordering behave differently:
```js
// note the `NOT` is not last:
await prisma.bugReproModel.findMany({
  where: {
    NOT: { tag: 'd' },
    topic: 'c',
  },
});

// RESULTS
[
  {
    "id": "65e2051871e92cf2cffdf22a",
    "topic": "b",
    "tag": "e"
  }
]

// now the `NOT` is the last field
await prisma.bugReproModel.findMany({
  where: {
    topic: 'c',
    NOT: { tag: 'd' },
  },
});

// RESULTS
[
  {
    "id": "65e2051971e92cf2cffdf22b",
    "topic": "c",
    "tag": "f"
  }
]
```

The results are different and only the 2nd one seems right.

Is this a bug, or is it documented behavior, or are we misunderstanding something?

### How to reproduce

Write a query with a `NOT` field, and more fields after the `NOT`. Then compare the results after moving the `NOT` field to the end.

### Expected behavior

We didn't expect the ordering of fields inside the `where` object to make a difference in the query.

### Prisma information

schema:
```prisma
model BugReproModel {
  id String @id @default(auto()) @map("_id") @db.ObjectId

  topic String
  tag   String
}
```

sample data:
```json
[
  { "id": "65e2051871e92cf2cffdf229", "topic": "a", "tag": "d" },
  { "id": "65e2051871e92cf2cffdf22a", "topic": "b", "tag": "e" },
  { "id": "65e2051971e92cf2cffdf22b", "topic": "c", "tag": "f" }
]
```

sample query:
```ts
// this one gave us unexpected results
await prisma.bugReproModel.findMany({
  where: {
    NOT: { tag: 'd' },
    topic: 'c',
  },
});

// this one seems ok
await prisma.bugReproModel.findMany({
  where: {
    topic: 'c',
    NOT: { tag: 'd' },
  },
});
```

### Environment & setup

OS: macOS
DB: mongoDB Atlas v6.0.14
Node.js: v18.14.2

### Prisma Version

npx prisma -v
```
Environment variables loaded from .env
prisma                  : 5.10.2
@prisma/client          : 5.10.2
Computed binaryTarget   : darwin
Operating System        : darwin
Architecture            : x64
Node.js                 : v18.14.2
Query Engine (Node-API) : libquery-engine 5a9203d0590c951969e85a7d07215503f4672eb9 (at node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Schema Engine           : schema-engine-cli 5a9203d0590c951969e85a7d07215503f4672eb9 (at node_modules/@prisma/engines/schema-engine-darwin)
Schema Wasm             : @prisma/prisma-schema-wasm 5.10.0-34.5a9203d0590c951969e85a7d07215503f4672eb9
Default Engines Hash    : 5a9203d0590c951969e85a7d07215503f4672eb9
Studio                  : 0.499.0
Preview Features        : tracing
```

**Comment by SevInf** [comment]
This was a bug in MongoDB connector, that was fixed in 5.11.0 (#22007).
Please update to the latest Prisma version and let us know if the problem is fixed for you. I am going to close this issue for now, but feel free to comment if update does not fix this and we'll reopen it.
Thank you!

**Comment by ehicks05** [comment]
Yes that fixed it! Thank you for the fix as well as the quick and clear response!
